crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -546,8 +542,50 @@ async fn execute_transactions(
 
     authority_state.transaction_manager.enqueue(txns).await?;
 
-    authority_state
+    let effects = authority_state
         .database
         .notify_read_effects(tx_digests)
-        .await
+        .await?;
+
+    assert_eq!(
+        effects.len(),
+        num_digests,
+        "Missing txn effects from transaction manager"
+    );
+    Ok(effects)
+}
+
+fn get_transactions(
+    digests: &Vec<TransactionDigest>,
+    authority_state: Arc<AuthorityState>,
+) -> SuiResult<Vec<VerifiedCertificate>> {
+    let synced_txns = authority_state
+        .database
+        .perpetual_tables
+        .synced_transactions
+        .multi_get(digests)?
+        .into_iter()
+        .flatten()
+        .map(|tx| tx.into());
+
+    let executed_txns = authority_state
```

Comment:
I believe if a certificate exists in the `certificates` table, the transaction must have executed and effects must exist. Do we need to read from this table?

Reply:
This is not quite the case - it is possible to have the cert but not the effects. The correct test for if a tx has been executed is to check if the effects exist.

Reply:
I thought this table is for executed certificates. Does sui write to this table in other cases?

Reply:
It is, but it in the case of a partially executed cert, you can have the certificate present but with effects missing.